File: src/alogorizm/sort/quicksort/QuickSort.java
package alogorizm.sort.quicksort;

import java.util.ArrayList;
import java.util.Random;
import java.util.stream.IntStream;

public class QuickSort {
    public ArrayList<Integer> sort(ArrayList<Integer> arrayList){
        if(arrayList.size()  <= 1){
            return arrayList;
        }
        int pivot = 0;
        ArrayList<Integer> leftArr = new ArrayList<>();
        ArrayList<Integer> rightArr = new ArrayList<>();
        for (int i = 1; i < arrayList.size(); i++) {
            if(arrayList.get(pivot) >= arrayList.get(i)){
                leftArr.add(arrayList.get(i));
            }else{
                rightArr.add(arrayList.get(i));
            }
        }
        System.out.println("leftArr : " + leftArr);
        System.out.println("rightArr : " + rightArr);

        ArrayList<Integer> mergeSortList = new ArrayList<>();
        mergeSortList.addAll(sort(leftArr));
        mergeSortList.add(arrayList.get(pivot));
        mergeSortList.addAll(sort(rightArr));
        System.out.println("mergeSort : " + mergeSortList);
        return mergeSortList;
    }

    public static void main(String[] args) {
        ArrayList<Integer> arrayList = new ArrayList<>();
        Random random = new Random();
        IntStream.rangeClosed(0, 10)
                .forEach(i -> arrayList.add(random.nextInt(100)));

        QuickSort quickSort = new QuickSort();
        ArrayList<Integer> sortList = quickSort.sort(arrayList);

    }
}

File: src/alogorizm/sort/quicksort/QuickSort.kt
package alogorizm.sort.quicksort

import kotlin.random.Random

fun quickSort(list: MutableList<Int>): MutableList<Int> {
    if (list.size <= 1) {
        return list
    }
    val pivot = 0
    val left = mutableListOf<Int>()
    val right = mutableListOf<Int>()

    for (i in 1 until list.size) {
        if (list[i] <= list[pivot]) {
            left.add(list[i])
        } else {
            right.add(list[i])
        }
    }
    println(left)
    println(right)

    val mergeSortList = mutableListOf<Int>()
    mergeSortList.addAll(quickSort(left))
    mergeSortList.add(list[pivot])
    mergeSortList.addAll(quickSort(right))
    return mergeSortList
}

fun main() {
    val list = Array(10) { Random.nextInt(0, 100) }.toMutableList()
    val sortList = quickSort(list)
    println(list)
    println(sortList)
}
